example-apps/my-nest-project/src/app.controller.ts

```diff
@@ -1,12 +1,25 @@
-import { Controller, Get } from '@nestjs/common';
+import { AuthAction, CreateAction } from '@kittgen/nestjs-authorization';
+import { Body, Controller, Put } from '@nestjs/common';
 import { AppService } from './app.service';
+import { IsAuthor } from './is-author.condition';
+import { CreateHelloDto as UpdateArticleDto } from './update-article.dto';
+
+export class ArticleAction {
+    static Read = CreateAction('read-article');
+    static Write = CreateAction('write-article');
+    static Admin = CreateAction('all');
+}
 
 @Controller()
 export class AppController {
   constructor(private readonly appService: AppService) {}
 
-  @Get()
-  getHello(): string {
+  @Put('/articles')
+  @AuthAction([
+    ArticleAction.Admin.if(ctx => ctx.switchToHttp().getRequest().body.published === true),
```

Comment:
a helper would be nice. maybe:

ReadAction.if(body(b => b.published === true))